opentelemetry/proto/collector/dynamicconfig/v1/dynamicconfig_service.proto

```diff
@@ -0,0 +1,123 @@
+}
+
+message ConfigRequest{
+  // Describes the version of the config service protocol to use. Required.
+  int32 proto_version = 1;
+
+  // The target resource. This resource pings the config server requesting
+  // updated configs, and the server responds with the config information.
+  opentelemetry.proto.resource.v1.Resource resource = 2;
+
+  // Optional. Identifies the last configuration fingerprint / version number.
+  // The fingerprint serves as a quick way for the config server to determine
+  // whether an update to the configs had occurred -- if the client request's
+  // fingerprint differs from the server's, then the server knows that a change
+  // had occured and an updated configuration should be sent.
+  int32 last_fingerprint = 3;
+
+  // An opaque string subject to individual implementation interpretation.
+  // Responsability to interpret and check safety of metadata falls on the
+  // specific implementation.
+  string metadata = 4;
+}
+
+message ConfigResponse {
+  // Describe the version of the config service protocol to use. Should match
+  // the request’s version.
+  int32 proto_version = 1;
+
+  // The fingerprint associated with the current config settings. Each change in
+  // configs yields a different fingerpting. The client may cache a fingerprint,
+  // then return it on the next request to enable the server to quickly check
+  // whether a change in the configs occured.
+  int32 fingerprint = 2;
+
+  // Configs specific to metric, namely the metric collection schedule
+  message MetricConfig {
+
+    // A schedule is used to apply a particular scheduling configuration to
+    // a metric. Order is important -- a metric that matches multiple
+    // schedules will take on the configurations specified in the first match.
+    message Schedule {
+
+      // Metrics with names that match the inclusion_pattern are targeted by
+      // this schedule. Metrics that match the exclusion_pattern are not
+      // targeted for this schedule, even if they match the inclusion pattern.
+      // A pattern should not exceded 120 characters.
+      string inclusion_pattern = 1;
+      string exclusion_pattern = 2;
+
+      // CollectionPeriod describes the sampling period for each metric. All
+      // larger units are divisible by all smaller ones.
+      enum CollectionPeriod {
+        NONE = 0;   // For non-periodic data. Client decides when to send points
+        SEC_1 = 1;
+        SEC_5 = 5;
+        SEC_10 = 10;
+        SEC_30 = 30;
+        MIN_1 = 60;
+        MIN_5 = 300;
+        MIN_10 = 600;
+        MIN_30 = 1800;
+        HR_1 = 3600;
+        HR_2 = 7200;
+        HR_4 = 14400;
+        HR_12 = 43200;
+        DAY_1 = 86400;
+        DAY_7 = 604800;
+      }
+      CollectionPeriod period = 3;
+
+      // Additional metadata associated with the schedule. Responsability to
+      // interpret and check safety of metadata falls on the specific
+      // implementation
+      string metadata = 4;
```

Comment:
Worth noting, we would likely want metadata to eventually be passed back from the agent into the collector via the [ExportMetricsServiceRequest](https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/collector/metrics/v1/metrics_service.proto#L35-L42) and from there passed into the exporter, to allow the exporter to demux requests and pass useful information back to the monitoring backend. For example, you could use this to allow writing to different backend endpoints, or different monitored resource types, or on different priority networking, or to include some other information useful to the monitoring backend.

Reply:
Gotcha, so would there still need to be a metadata field associated with ConfigRequest? Or would it be just in ExportMetricsServiceRequest?

Reply:
Let's expand on this.

"Optional. Additional metadata associated with the schedule that may benefit the metric backend when writing data to the metric backend. A particular exporter may store this metadata and piggyback to the backend to make the collection more efficient. Examples include (1) preferred priority in terms of QoS (2) a more concise representation of the collection schedule and applicable metrics. ..."